clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -94,8 +94,8 @@ public class ProducerConfig extends AbstractConfig {
                                                  + "batch size is under this <code>batch.size</code> setting.";
 
     /** <code>partitioner.adaptive.partitioning.enable</code> */
-    public static final String PARTITIONER_ADPATIVE_PARTITIONING_ENABLE_CONFIG = "partitioner.adaptive.partitioning.enable";
```

Comment:
This is a public field in a public class: https://kafka.apache.org/35/javadoc/org/apache/kafka/clients/producer/ProducerConfig.html#PARTITIONER_ADPATIVE_PARTITIONING_ENABLE_CONFIG 

Many customers may have code which may be relying on this typo and this change will break those customers. There is very little upside in changing it now and huge downside wrt backward compatibility. I will suggest to keep it as it is for now. We will have to live with it until next major version (4.x).

Reply:
If we want to remove (rename) this field in a future version, we can deprecate it now and introduce a new field with the correct spelling.

That might fall into KIP territory, though. @divijvaidya thoughts?

Reply:
Thanks for your suggestion, that is the right way to go about it @C0urante. I understand that a KIP for this change might be a an overdo but let's follow the rules decided by the community (and create a KIP for deprecation).